### PR TITLE
Logout

### LINKED ISSUE #126: User Story 13, User can log out
```
As a registered user, merchant, or admin
When I visit the logout path
I am redirected to the welcome / home page of the site
And I see a flash message that indicates I am logged out
Any items I had in my shopping cart are deleted
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #130

### PR DESCRIPTION
**Pull request by tonyr729**
Resolves #126

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -16,4 +15,10 @@ def create
       render :new
     end
   end
+
+  def destroy
+    session.clear
+    
+    redirect_to root_path
```
Comment (nicovigil1) [inline app/controllers/sessions_controller.rb]: picasso would be proud

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def create
   end
 
   def show
-    @user = User.find(session[:user_id])
+    @user = current_user || User.find(session[:user_id])
```
Comment (nicovigil1) [inline app/controllers/users_controller.rb]: You are a god among men
